JsonSchema/CachedSchemaStorage.php

```diff
@@ -0,0 +1,153 @@
+<?php
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\ValidationBundle\JsonSchema;
+
+use InvalidArgumentException;
+use JsonSchema\Entity\JsonPointer;
+use JsonSchema\Iterator\ObjectIterator;
+use JsonSchema\SchemaStorage;
+use Sulu\Bundle\ValidationBundle\Exceptions\MalFormedJsonException;
+use Symfony\Component\Config\ConfigCache;
+use Symfony\Component\Config\Exception\FileLocatorFileNotFoundException;
+use Symfony\Component\Config\FileLocatorInterface;
+use Symfony\Component\Config\Resource\FileResource;
+
+class CachedSchemaStorage extends SchemaStorage
+{
+    const FILE_PREFIX = 'file://';
+
+    /**
+     * @var FileLocatorInterface
+     */
+    private $fileLocator;
+
+    /**
+     * @var array
+     */
+    private $configuredSchemas;
+
+    /**
+     * @var bool
+     */
+    private $debugMode;
+
+    /**
+     * @var string
+     */
+    private $cacheFilePath;
+
+    /**
+     * @param array                $configuredSchemas array containing all file paths to configured schemas
+     * @param FileLocatorInterface $fileLocator
+     * @param string               $cacheFilePath
+     * @param string               $environment
+     */
+    public function __construct(
+        array $configuredSchemas,
+        FileLocatorInterface $fileLocator,
+        $cacheFilePath,
+        $environment
+    ) {
+        parent::__construct();
+
+        $this->fileLocator = $fileLocator;
+        $this->configuredSchemas = $configuredSchemas;
+        $this->debugMode = $environment !== 'prod';
+        $this->cacheFilePath = $cacheFilePath;
+    }
+
+    public function initializeCache()
```

Comment:
this should be called directly before needed - introduce a `private $initialized = false` which will be checked and set in this function.